src/torcontrol.cpp

```diff
@@ -414,6 +416,12 @@ TorController::TorController(struct event_base* _base, const std::string& _targe
         LogPrint(BCLog::TOR, "tor: Reading cached private key from %s\n", GetPrivateKeyFile());
         private_key = pkf.second;
     }
+    // Set Single Onion Service flag
+    if (mode == 1)
```

Comment:
Having `-listenonion=<n>` flag requires allowed values checks and isn't obvious enough as for anonymous/non-anonymous service option. Maybe it would be better to just add another boolean flag instead, with some obvious name?